pool/postgres.go

```diff
@@ -243,8 +244,15 @@ func decodeHashDataRows(rows *sql.Rows) ([]*HashData, error) {
 			return nil, errs.DBError(errs.Decode, desc)
 		}
 
-		hashData := &HashData{uuid, accountID, miner, ip, hashRate, updatedOn}
-		toReturn = append(toReturn, hashData)
+		hashRat, ok := new(big.Rat).SetString(hashRate)
+		if !ok {
+			desc := fmt.Sprintf("%s: unable to decode big.Rat string: %v",
+				funcName, err)
+			return nil, errs.DBError(errs.Parse, desc)
+		}
+
+		hashData := &HashData{uuid, accountID, miner, ip, hashRat, updatedOn}
+		toReturn[uuid] = append(toReturn[uuid], hashData)
```

Comment:
I think the key on this map is supposed to be the accountID, not the hashData ID.

Reply:
good catch